components/openweather_api/actions/get-current-weather-by-location/get-current-weather-by-location.mjs

```diff
@@ -0,0 +1,73 @@
+import { ConfigurationError } from "@pipedream/platform";
+import openweatherApi from "../../openweather_api.app.mjs";
+
+export default {
+  key: "openweather_api-get-current-weather-by-location",
+  name: "Get current weather by location",
+  description: `Retrieves the current weather condition by location longitude and latitude. 
+  [See the docs here](https://openweathermap.org/current). For more accurate reading, you are adviced to fill in the stateccode and/or country code`,
+  version: "0.0.1",
+  type: "action",
+  props: {
+    openweatherApi,
+    appid: {
+      propDefinition: [
+        openweatherApi,
+        "appId",
+      ],
+    },
```

Comment:
It looks like the `appId` or api key is provided in `this.openweatherApi.$auth.api_key`, so this prop isn't necessary.
```suggestion
```

Reply:
More clarification is needed here.  
I am meant to pass an ID which I got from my profile in open weather. How do i access the appId property from this.openweatherApi.$auth.api_key.

Reply:
When you set up your openweather account in Pipedream giving Pipedream access to your account, you had to enter in the api_key. Pipedream keeps that. Try a `console.log` from the `run` method to see what Pipedream has stored in `$auth`.
`console.log(this.openWeatherApi.$auth);`
You should get an object with a key `api_key`, and a value containing the actual api key.

Reply:
Thanks for the clarification